pkg/canary/deployment_controller.go

```diff
@@ -242,9 +246,7 @@ func (c *DeploymentController) createPrimaryDeployment(cd *flaggerv1.Canary) err
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      primaryName,
 				Namespace: cd.Namespace,
-				Labels: map[string]string{
-					label: primaryLabelValue,
-				},
+				Labels:    labels,
```

Comment:
This filtering only applies to the deployment labels. Should it apply to the spec.template.metadata.labels as well?
From what I understood this is only to solve a very specific use case so I'm not sure if it should apply everywhere.